airflow/plugins/operators/realtime_to_flattened_json_operator.py

```diff
@@ -0,0 +1,218 @@
+                parsed = parse_pb(feed_fname, logger=logger)
+
+                if parsed and "entity" in parsed:
+                    for record in parsed["entity"]:
+                        record["header"] = parsed["header"]
+                        record["calitp_filepath"] = str(parsed["calitp_filepath"])
+                        record["calitp_itp_id"] = int(feed[0])
+                        record["calitp_url_number"] = int(feed[1])
+                        gzipfile.write((json.dumps(record) + "\n").encode("utf-8"))
+                        written += 1
+
+        if not written:
+            logger.warning("did not parse any entities, skipping upload")
+            return
+
+        num_bytes = os.stat(gzip_fname).st_size
+        logger.info(
+            f"writing {written} lines ({num_bytes} bytes) from {gzip_fname} to {dst_path + google_cloud_file_name}"
+        )
+        fs.put(
+            gzip_fname, dst_path + google_cloud_file_name,
+        )
+        logger.info(
+            "took {} seconds to process {} files".format(
+                (datetime.now() - start).total_seconds(), len(all_files)
+            )
+        )
+
+
+def execute(context, filename_prefix, rt_file_substring, src_path, dst_path):
+    # get execution_date from context:
+    # https://stackoverflow.com/questions/59982700/airflow-how-can-i-access-execution-date-on-my-custom-operator
+    iso_date = str(context["execution_date"]).split("T")[0]
+
+    # fetch files ----
+    feed_files = fetch_bucket_file_names(src_path, rt_file_substring, iso_date)
+
+    # gcfs does not seem to play nicely with multiprocessing right now
+    # https://github.com/fsspec/gcsfs/issues/379
+    with ThreadPoolExecutor(max_workers=4) as pool:
+        args = [
+            (i, feed, files, filename_prefix, iso_date, dst_path)
+            for i, (feed, files) in enumerate(feed_files.items())
+        ]
+        list(pool.map(handle_one_feed, *zip(*args)))
+
+
+class RealtimeToFlattenedJSONOperator(BaseOperator):
+    @apply_defaults
+    def __init__(
+        self,
+        header_details,
+        entity_details,
+        filename_prefix,
+        rt_file_substring,
+        cast={},
+        is_timestamp=[],
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        # could not just set self.fs here - ran into this issue:
+        # https://stackoverflow.com/questions/69532715/google-cloud-functions-using-gcsfs-runtimeerror-this-class-is-not-fork-safe
+        # self.fs = get_fs()
+        self.header_details = header_details
+        self.entity_details = entity_details
+        self.filename_prefix = filename_prefix
+        self.rt_file_substring = rt_file_substring
+        self.cast = cast
+        self.time_float_cast = {col: "float" for col in is_timestamp}
+        self.src_path = f"{get_bucket()}/rt/"
+        self.dst_path = "".join(
+            [
+                f"{get_bucket()}/rt-processed_test_2022-01-24/",
```

Comment:
We should remove the testing bucket path here

Reply:
I think we need a conversation around this in general; IMO paths/keys should largely be fixed between production and testing/staging, but then the bucket should differ.